Explain accessing dock widget wrappers (#892)
napari/docs: docs/plugins/advanced_topics/widget_communication.md

## Access a widget by name with `viewer.window.dock_widgets`

If the goal is to access a target widget, without creating a widget, use the `dock_widgets` property.
The `dock_widgets` property provides access to a read-only mapping of all docked widgets in the viewer. You can list available widget names with `viewer.window.dock_widgets.keys()`.

This public API returns the (inner) widget, not the `QtViewerDockWidget` wrapper.

> To control the dock widget itself, access the `QtViewerDockWidget` wrapper using the `.parent()` method on the returned `QWidget`, or `.native.parent()` for magicgui `Widget`.

Reviewer: ```suggestion
To access the `QtViewerDockWidget` wrapper, use the `.parent()` method on the returned `QWidget`, or `.native.parent()` for a magicgui `Widget`.
```

For example, to programmatically show the "Home" dock widget:

```python
widget = viewer.window.dock_widgets["Home"]
qt_widget = widget.native if hasattr(widget, "native") else widget
dock_widget = qt_widget.parent()  # Get the QtViewerDockWidget wrapper
dock_widget.show()
dock_widget.raise_()
```

```{important}
Do not use `viewer.window._dock_widgets` to access `QtViewerDockWidget`.

> This is a private API that may be removed in any release.

Reviewer: ```suggestion
This is a private API that will be removed soon.
```
